### PR TITLE
feat: [DHIS2-10948] fetch profile widget

### PR DESCRIPTION
**Pull request by huocha**

### CODE REVIEW

File: src/core_modules/capture-ui/FlatList/FlatList.component.js
Comment (simonadomnisoru) [inline src/core_modules/capture-ui/FlatList/FlatList.component.js:11]: Hey, 
What do you think about using the constant `spacersNum.dp16` from `@dhis2/ui` instead of the number 16 to help maintain the homogeneity of the stypes?
Thank you!

File: src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js:20]: Hey @huocha , 

I would advise to move the `useSelector` logic in the parrent of `ProfileWidgetPlain`.  You can pass` teiId and programId` as props to `ProfileWidgetPlain`. This way we ensure there is no redux logic in the WidgetProfile. 
What do you think? Thanks! 😃

File: src/core_modules/capture-ui/FlatList/FlatList.component.js
Comment (huocha) [reply src/core_modules/capture-ui/FlatList/FlatList.component.js]: @simonadomnisoru totally makes sense! I will change that

File: src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js
Comment (huocha) [reply src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js]: @simonadomnisoru I agree, I was thinking about moving that but forgot. just changed that :)

File: cypress/integration/EnrollmentPage/WidgetProfile/index.js
Comment (simonadomnisoru) [inline cypress/integration/EnrollmentPage/WidgetProfile/index.js:7]: Hey, 
Do you think is more sensible to check in this step for the texts fields that are displayed inside the`FlatList` component? 
Thanks!

Comment (huocha) [reply cypress/integration/EnrollmentPage/WidgetProfile/index.js:7]: Hi @simonadomnisoru I do think about it but the fields can be different between based on the api version. I think better leave it like this for now and come back after the implementation of edition

**Review by simonadomnisoru — APPROVED** [review]

**Review by JoakimSM — COMMENTED** [review]
Hey @huocha, I think this looks good overall, so please don't be discouraged by the number of comments I've made 🙂 I feel it's mostly about us getting on the same page, I think it's natural to have some things to discuss in the first PRs.

Please don't hesitate to reach out if you want to discuss something in more detail (slack chat or meet).

File: src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js
Comment (huocha) [reply src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js:43]: I thought about it before but couldn't find any example, in the docs it seems that we can only pass a single query https://runtime.dhis2.nu/#/hooks/useDataQuery

File: src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPageDefault/EnrollmentPageDefault.container.js
Comment (huocha) [reply src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPageDefault/EnrollmentPageDefault.container.js:14]: Yes, absolutely makes sense :)

File: src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js
Comment (huocha) [reply src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js:77]: In case we want to display the image (for the photograph) I think it's better to create the element here to keep the flatlist generic. 
Another approach is to pass the value to the flat list and create the element there. What do you prefer?

File: src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js
Comment (huocha) [reply src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js:66]: I see, will change that

File: src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js
Comment (JoakimSM) [reply src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js]: Ok, we can experiment with this at a later time and leave it as it is for now 🙂

File: src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js
Comment (JoakimSM) [reply src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js]: My preferred approach to this would be to allow for what flow defines as Node (could be number, text, react element etc). Anything defined as Node can be rendered by React (hence, node is the return type for the functional components and the return type of the render function of the class component). https://flow.org/en/docs/react/types/#toc-react-node

For images the convertMethod will return a react element itself.

**Review by JoakimSM — APPROVED** [review]
Thanks for your effort on this and putting up with all my comments. Looks good 👍 

Im approving now, but it would be great if you look into why the cypress test is failing before you merge.

**Comment by huocha** [comment]
@JoakimSM thank you for the comment. The test seems flaky since it resolved after I merge master.

**Pull request merged**
